File: packages/react-native/ReactAndroid/src/main/java/com/facebook/react/fabric/CoreComponentsRegistry.java
/*
 * Copyright (c) Meta Platforms, Inc. and affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

package com.facebook.react.fabric;

import com.facebook.infer.annotation.Nullsafe;
import com.facebook.jni.HybridData;
import com.facebook.proguard.annotations.DoNotStrip;

@Nullsafe(Nullsafe.Mode.LOCAL)
@DoNotStrip
public class CoreComponentsRegistry {
  static {
    FabricSoLoader.staticInit();
  }

  @DoNotStrip private final HybridData mHybridData;

  @DoNotStrip
  private native HybridData initHybrid(ComponentFactory componentFactory);

  @DoNotStrip
  private CoreComponentsRegistry(ComponentFactory componentFactory) {
    mHybridData = initHybrid(componentFactory);
  }

  @DoNotStrip
  public static CoreComponentsRegistry register(ComponentFactory componentFactory) {
    return new CoreComponentsRegistry(componentFactory);
  }
}

File: packages/react-native/ReactAndroid/src/main/java/com/facebook/react/fabric/CoreComponentsRegistry.kt
/*
 * Copyright (c) Meta Platforms, Inc. and affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

package com.facebook.react.fabric

import com.facebook.jni.HybridData
import com.facebook.proguard.annotations.DoNotStripAny

@DoNotStripAny
public class CoreComponentsRegistry private constructor(componentFactory: ComponentFactory) {

  @Suppress("NoHungarianNotation")
  private val mHybridData: HybridData = initHybrid(componentFactory)

  private external fun initHybrid(componentFactory: ComponentFactory): HybridData

  public companion object {
    init {
      FabricSoLoader.staticInit()
    }

    @JvmStatic
    public fun register(componentFactory: ComponentFactory): CoreComponentsRegistry =
        CoreComponentsRegistry(componentFactory)
  }
}
